### PR TITLE
Add Library endpoints

### PR DESCRIPTION
**Pull request by henryc132**

### CODE REVIEW

File: Server/App/api/library.py [hunk 1]
```diff
@@ -29,15 +61,59 @@ class LibraryCopyHandler(BaseHandler):
...
 
 
 class LibraryDeckRemoveHandler(BaseHandler):
     @authorize_request_and_create_db_connector
     @extract_user_id
     def delete(self, user_id, connector, uuid):
         """Remove a deck from a user's library and delete it if they are the owner."""
-        self.abort(404)
+
+        connector.call_procedure('LIBRARY_REMOVE', user_id, uuid)
```
Comment (ClaytonPassmore) [inline Server/App/api/library.py:114]: If the deck is owned by the user that is deleting it, we also have to mark the deck as deleted and delete all of its cards and tags

**Comment by ClaytonPassmore** [comment]
Overall looks great! Just one case we need to cover where the user deleting the deck also owns the deck.

File: Server/App/api/library.py (hunk 1 above)
Comment (ClaytonPassmore) [reply Server/App/api/library.py:114]: Also need to remove the needreview entries.

**Comment by henryc132** [comment]
@ClaytonPassmore Updated the library_remove procedure

**Review by ClaytonPassmore — APPROVED** [review]
Cool, looks good!

**Pull request merged**